From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rob Wu <rob@robwu.nl>
Date: Tue, 5 Nov 2024 21:24:20 +0000
Subject: [PATCH] Bug 1926140 - Replace pipes imports r=jmaher

pipes does not exist in Python 3.11 any more

Differential Revision: https://phabricator.services.mozilla.com/D227964
---
 js/src/tests/lib/results.py                | 4 ++--
 testing/mozbase/mozdevice/mozdevice/adb.py | 3 ---
 testing/xpcshell/runxpcshelltests.py       | 6 +++---
 3 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/js/src/tests/lib/results.py b/js/src/tests/lib/results.py
index f90206035427..af580a9ecb47 100644
--- a/js/src/tests/lib/results.py
+++ b/js/src/tests/lib/results.py
@@ -1,15 +1,15 @@
 import json
-import pipes
 import re
+import shlex
 
 from .progressbar import NullProgressBar, ProgressBar
 from .structuredlog import TestLogger
 
 # subprocess.list2cmdline does not properly escape for sh-like shells
 
 
 def escape_cmdline(args):
-    return " ".join([pipes.quote(a) for a in args])
+    return " ".join([shlex.quote(a) for a in args])
 
 
 class TestOutput:
diff --git a/testing/mozbase/mozdevice/mozdevice/adb.py b/testing/mozbase/mozdevice/mozdevice/adb.py
index cf1b78064e20..be0f65f1426e 100644
--- a/testing/mozbase/mozdevice/mozdevice/adb.py
+++ b/testing/mozbase/mozdevice/mozdevice/adb.py
@@ -4,7 +4,6 @@
 
 import io
 import os
-import pipes
 import posixpath
 import re
 import shlex
@@ -1294,8 +1293,6 @@ class ADBDevice(ADBCommand):
         """Utility function to return quoted version of command argument."""
         if hasattr(shlex, "quote"):
             quote = shlex.quote
-        elif hasattr(pipes, "quote"):
-            quote = pipes.quote
         else:
 
             def quote(arg):
diff --git a/testing/xpcshell/runxpcshelltests.py b/testing/xpcshell/runxpcshelltests.py
index 8ef8aed65f69..2002f122b43e 100755
--- a/testing/xpcshell/runxpcshelltests.py
+++ b/testing/xpcshell/runxpcshelltests.py
@@ -7,9 +7,9 @@
 import copy
 import json
 import os
-import pipes
 import random
 import re
+import shlex
 import shutil
 import signal
 import subprocess
@@ -353,11 +353,11 @@ class XPCShellTestThread(Thread):
         )
         self.log.info("%s | environment: %s" % (name, list(changedEnv)))
         shell_command_tokens = [
-            pipes.quote(tok) for tok in list(changedEnv) + completeCmd
+            shlex.quote(tok) for tok in list(changedEnv) + completeCmd
         ]
         self.log.info(
             "%s | as shell command: (cd %s; %s)"
-            % (name, pipes.quote(testdir), " ".join(shell_command_tokens))
+            % (name, shlex.quote(testdir), " ".join(shell_command_tokens))
         )
 
     def killTimeout(self, proc):
